gapis/api/gles/gles.api

```diff
@@ -732,21 +750,27 @@ sub void ApplyDynamicContextState(ref!Context ctx, ref!DynamicContextState dynam
     stencil_id := backbuffer.StencilAttachment.Renderbuffer.ID
 
     color_buffer := ctx.Objects.Renderbuffers[color_id]
-    color_buffer.Width = dynamicState.BackbufferWidth
-    color_buffer.Height = dynamicState.BackbufferHeight
-    color_buffer.InternalFormat = dynamicState.BackbufferColorFmt
+    color_buffer.Image = new!Image(
```

Comment:
Do we not get a `DynamicContextState` whenever we rebind a context?

If so, aren't you nuking the image data?

Reply:
1) We never actually set the data, so it will be always empty.
2) Nuking seems like a better option then leaving in data of potentially different renderbuffer, or potentially different size.

The better solution would be to do the same thing we do for textures and set the Data as "GPU produced" at this point.

Reply:
> 1. We never actually set the data, so it will be always empty

This is on my TODO list, hence why I brought it up.

> Nuking seems like a better option then leaving in data of potentially different renderbuffer, or potentially different size.

Sure - but only if the framebuffer actually changes dimensions or format. Currently I don't think consider that logic.
Does DCE not use the data field as a key?

Reply:
Even if the dimensions and format is same, the underlying renderbuffer may be different.

Reply:
Fair point. Perhaps we should only be generating new `DynamicContextState`s if the `EGLDisplay` or `EGLSurface`s change?

I have no idea if they can change dimensions / formats however.

Reply:
Yeah, good questions.  It is getting a bit of topic with respect to this PR though - as that just wraps the Data,Width,Height,Format fields in ref!Image, and the Data is still always nil for the default framebuffer.